### ISSUE TITLE
how these statistical values are calculated

### ISSUE THREAD

**Issue by jack1234lee** [robbiewatt1/ClimateDiffuse#4]
Hello. I would like to know how these statistical values in DatasetUS.py (show the corresponding screenshot of statistical values) are calculated.Thank you very much for your time and assistance.
Best regards
![image](https://github.com/user-attachments/assets/43efa586-02a4-4e74-b2b7-bc91cb12abb0)

**Comment by lm2612** [comment]
Hi and thanks for the question. 

These values are the mean and standard deviation calculated across the entire training dataset for each variable (three variables temp, u, v). 

The easiest way to do this is to create an instance of the UpscaleDataset with no normalization where the means are 0. (ie. subtract zero) and standard deviations are 1. (ie scale by 1) i.e.:
```
dataset_raw = UpscaleDataset(
    datadir, 
    year_start=1950, 
    year_end=2017, 
    normalize_rawdata_mean = torch.Tensor([0., 0., 0.]), 
    normalize_rawdata_std = torch.Tensor([1., 1., 1.]),
    normalize_residual_mean = torch.Tensor([0., 0., 0.]), 
    normalize_residual_std = torch.Tensor([1., 1., 1.])
    )
```
Now you can access the raw inputs with `dataset_raw.inputs` and the raw residuals with `dataset_raw.targets`. These are torch tensors, so you can just take the mean and standard deviation across these (just make sure to check the dimensions to take the mean/std separately for the three variables T, u and v).

Best wishes,
Laura

**Comment by jack1234lee** [comment]
Thank you very much for your reply. However, I still have some doubts. The means and variances of the training set and the validation set are different. Why is it assumed in the code that their means and variances are the same? Thank you very much!

**Comment by lm2612** [comment]
What numbers are you finding for the means/variances of training and validation set? It is true that we might expect small differences in A) the mean, because our training set is 1950-2017 while our test set is 2018-2022, which would have been warmer on average and B) the variance for the same reasons and because our test set is much smaller than our training set. 

But in a real life scenario where we are trying to train a model on past historical data and predict in the future, we wouldn't have access to the "future" statistics (i.e., the means/variances in the test set 2018-2022). So we are doing the best we can by using the same statistics of the past. We are trying to keep the problem as similar as possible to a real timeseries problem (train on the past, predict on the future) - the fact that the climate is changing is a general challenge for AI. 

Let me know if you have anymore thoughts or questions!

**Comment by jack1234lee** [comment]
I see! Your answer has enlightened me. Thank you so much!Best wishes!
